python.md

```diff
@@ -0,0 +1,1376 @@
+```
+
+To set up a package to be upgraded or installed, you can use code like the
+following:
+
+```python
+h = read_rpm_header(ts, sys.argv[1])
+ts.addInstall(h, sys.argv[1], "u")
+```
+
+This example expects a package file name on the command line (accessed with
+`sys.argv[1]`), and reads in the package header using the `readRpmHeader`
+function introduced previously.
+
+The call to `addInstall` adds the header object (and the associated RPM package
+file) for an upgrade with the `'u'` mode flag. The name of the package file,
+from `sys.argv[1]`, is passed as the arbitrary data for the transaction set run
+callback function.
+
+
+## Transaction elements
+
+Transaction sets are made up of transaction elements. A transaction element
+makes up one part of a transaction and holds one package per operation (install
+or remove) in each transaction set. That is, there is one transaction element
+per package per operation in the transaction set. You can iterate over a
+transaction set to get each transaction element. Once you have a transaction
+element, you can call methods on each element to check entries in the header as
+well as get dependency sets for the package.
+
+The following table lists the informational methods you can call on a
+transaction element. Most of the listed methods return a single value.
+
+Table 17-4 Informational methods on transaction sets
+
+| Method | Returns                                            |
+|--------|----------------------------------------------------|
+| `A`    | Returns package architecture                       |
+| `E`    | Returns package epoch                              |
+| `O`    | Returns package operating system                   |
+| `R`    | Returns package release number                     |
+| `V`    | Returns package version                            |
+| `N`    | Returns package name                               |
+| `NEVR` | Returns package name-epoch-version-release         |
+| `DS`   | Returns the package dependency set for a given tag |
+| `FI`   | Returns the file info set for the package          |
+
+For more complex checking, the DS method returns the package dependency set for
+a given tag:
+
+```python
+ds = te.DS(tag_name)
+```
+
+Pass one of `'Providename'`, `'Requirename'`, `'Obsoletename'`, or
+`'Conflictname'` for the tag name. For example:
+
+```python
+ds = te.DS('Requirename')
+```
+
+The `Files` method returns the file info set for the package:
+
+```python
+for f in te.Files(tag_name):
+    print(f.name)
+```
+
+This example shows how to iterate through a transaction set to get transaction
+elements.
+
+```python
+#!/usr/bin/python
+# Adds all package files on command line to a transaction
+# and prints out the transaction elements.
+# Usage:
+# python te.py rpm_file1.rpm rpm_file2.rpm ...
+
+import os
+import sys
+import rpm
+
+def read_rpm_header(ts, filename):
+    fd = os.open(filename, os.O_RDONLY)
+    try:
+        h = ts.hdrFromFdno(fd)
```

Comment:
You also just pass a file name directly to hdrFromFdno():

```
>>> h = ts.hdrFromFdno('/tmp/setup-2.15.0-8.fc41.noarch.rpm')
>>> h['name']
'setup'
```

...or a Python file object...